### PR TITLE
Style : Improving Viewer Landing

### PR DESCRIPTION
**Pull request by ingrid011**
**Description**
This solved issue #14 
Based on team discussion, viewer landing page needs to be improved especially on the styling part.
Kindly read below implementations that are done.
**NOTES: some of the codes (example, using `properties` instead of `props`, or using `Number.parseInt` instead of `parseInt`, etc) are adjusted because there're some `unicorn` rules error.**

**Implementation**
1. Refolder viewer landing page to be on `/streaming/preview/{streamid}` & rename it as `streamid` instead (commit [f5c2a40](https://github.com/inlivedev/live-streaming-essential-kit/pull/15/commits/f5c2a40950f5c4b64b6d8a28ef2c65e6e08b7989))
2. Since most of the descriptions texts from endpoint streams are relatively short sentence (eg. 3 text only), then we align text to `center` (both mobile & web)  (commit [7f8bacd](https://github.com/inlivedev/live-streaming-essential-kit/pull/15/commits/7f8bacdb77218f9d68702734f223c94e253f845d))
3. Revised the background viewer landing styling based on updated figma design (commit [e677942](https://github.com/inlivedev/live-streaming-essential-kit/pull/15/commits/e677942d9b86a4de32db252bbbc5178887dee4fb)).
    
4. Also, revised the username form (commit [93833d2](https://github.com/inlivedev/live-streaming-essential-kit/pull/15/commits/93833d2b30099d8e3e643873f92a8741ec79e70b)) : 
    - To be displayed if stream status is live / ended
    - Add functionality to enable/disable submit username button if stream status is live 
Screen Shot 2022-10-17 at 19 58 52
Screen Shot 2022-10-18 at 08 16 09

- If stream status is ended, then the button just disable even though user type in input field
Screen Shot 2022-10-18 at 08 15 15

5. For now since the schedule status not functioning "Notify Me" button, so I take it out again (commit [f83d762](https://github.com/inlivedev/live-streaming-essential-kit/pull/15/commits/f83d762242238fdf86f4978f12e5781c38f8e8b1))

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -21,7 +21,8 @@ module.exports = {
   rules: {
     camelcase: ['error', { properties: 'always' }],
     eqeqeq: ['error', 'always'],
-    'unicorn/prefer-ternary': ['error', 'only-single-line']
+    'unicorn/prefer-ternary': ['error', 'only-single-line'],
+    'unicorn/template-indent': 'off'
```
Comment (ingrid011) [inline .eslintrc.cjs:25]: while working on itenary condition on lit element (on the render() side part), prettier & unicorn/template-indent rules are conflicted at each other (see https://stackoverflow.com/questions/56337176/prettier-and-eslint-indents-not-working-together).

I can't commit my work if not solving this. If I solved prettier problem, then the unicorn/template-indent error will show up. If solved the unicorn/template-indent problem, then the prettier error will show up.

For now, I just turn it off. Kindly advice if it's okay.

File: src/features/viewer/landing/app-viewer-landing.js [hunk 2]
```diff
@@ -53,11 +76,18 @@ export class AppViewerLanding extends LitElement {
...
+              <button type="button" disabled class="notify-me-button disable">
+                Notify Me
+              </button>
+            `
+          : html`
+              <viewer-join-live
+                streamId=${this.streamId}
+                streamStatus=${this.streamStatus}
+              ></viewer-join-live>
+            `}
```
Comment (ingrid011) [inline src/features/viewer/landing/app-viewer-landing.js:90]: if stream scheduled then will show disabled notify me button (just a UI), else, it still render the input username form

File: src/features/viewer/landing/viewer-join-live.js [hunk 3]
```diff
@@ -157,21 +176,21 @@ export class ViewerJoinLive extends LitElement {
...
-    const username = form ? this.checkValue(form['username'].value) : null;
+    const username = form ? this.checkValue(form['username'].value) : undefined;
 
     if (username !== undefined && username !== null) {
       // save the username into local storage
       localStorage.setItem('viewer-username', username);
 
       // go to the next page
-      return window.location.replace('/watch-live-stream/' + this.streamId);
+      return window.location.replace('/streaming/watch/' + this.streamId);
```
Comment (ingrid011) [inline src/features/viewer/landing/viewer-join-live.js:193]: set as per agreed pages name on our [slack conversation](https://asumsimedia.slack.com/archives/C043RVBMJNB/p1665978852923509).

File: src/features/viewer/landing/viewer-join-live.js [hunk 4]
```diff
@@ -185,15 +204,27 @@ export class ViewerJoinLive extends LitElement {
               class="input-field ${this.showError ? 'red' : 'gray'}"
               type="text"
               name="username"
+              id="user"
               placeholder="Set username to watch"
               minlength="3"
+              @keyup=${(/** @type {{ target: { value: any; }; }} */ event) =>
+                this.enableDisableButton(event.target.value)}
```
Comment (ingrid011) [inline src/features/viewer/landing/viewer-join-live.js:211]: in order can detect if user input text on the input field, then enable the button

**Comment by ingrid011** [comment]
Please recheck updated [above description](https://github.com/inlivedev/live-streaming-essential-kit/pull/15#issue-1411002648).

**Comment by ingrid011** [comment]
Please recheck updated [above description](https://github.com/inlivedev/live-streaming-essential-kit/pull/15#issue-1411002648).

File: .eslintrc.cjs (hunk 1 above)
Comment (faiq-naufal) [reply .eslintrc.cjs:25]: hmm, I think this is fine. I think we should prefer the prettier one

**Comment by faiq-naufal** [comment]
@ingrid011 Because we will access this page through URL `/streaming/preview/{streamid}`, I think better to change all `landing` words into `preview` words to make it consistent. Don't forget the folder and file name too

File: src/features/viewer/landing/viewer-join-live.js (hunk 3 above)
Comment (faiq-naufal) [reply src/features/viewer/landing/viewer-join-live.js:196]: window.location.replace will remove the current URL (the preview URL) from the document history which is not ideal. Consider to use `window.location.href = '/streaming/watch/${this.streamId}'`

**Comment by ingrid011** [comment]
> @ingrid011 Because we will access this page through URL `/streaming/preview/{streamid}`, I think better to change all `landing` words into `preview` words to make it consistent. Don't forget the folder and file name too

@faiq-naufal make sense mas, thanks for the advice. Done at commit [56d19c2](https://github.com/inlivedev/live-streaming-essential-kit/pull/15/commits/56d19c227710b174af70de64cc999917077700d6), kindly help to recheck it again.

**Review by azhsetiawan — APPROVED** [review]

File: src/features/viewer/landing/viewer-join-live.js (hunk 3 above)
Comment (ingrid011) [reply src/features/viewer/landing/viewer-join-live.js]: @faiq-naufal done mas at commit [6908a5a](https://github.com/inlivedev/live-streaming-essential-kit/pull/15/commits/6908a5ae20f88367c9102ea4a20465ac5cf21281). Thank you

**Review by faiq-naufal — APPROVED** [review]
LGTM

**Pull request merged**